### ISSUE TITLE
IS CHICKENBONES RETIRED?

### ISSUE THREAD

**Issue by chloeprince** [Chicken-Bones/NotEnoughItems#312]
Sorry to Leave this message here, But I am not sure how else to contact you... its been a awhile since you've had activity here, and I was wondering if CB is retired and someone else has taken over NEI?

It needs update for 1.7.10 version to handle the latest Forge Versions beyond 1448.  

I looks like CB is starting work on 1.8 MC, but if he isn't going to update NEI and other mods for anything higher than Forge 1448, maybe someone else could?  

NEI doesn't render its TABS in creative mode correctly, and it also glitches out on many of the items graphics in the NEI slots.  I've tried it with the recommended version of Forge Currently 1558, and also on the new 1614 version, and NEI just doesn't work right... which sucks, because I can't update my Mod Pack to the latest Forge on my server, because then NEI won't work.

Can anyone help?

**Comment by GirafiStudios** [comment]
He's busy from what I have heard.

And no, somebody else can not legally update NEI without permissions from Chicken Bones.

**Comment by Chicken-Bones** [comment]
Indeed, I am currently very busy, and it's a shame that I haven't been able
to find time to contribute more to the community.
On the contrary, the MIT license permits you to make a copy and do whatever
you wish with it.
If someone (or some group) would be willing to take over the maintenance of
NEI and other mods, to do ports, bugfixes, or feature additions, I would be
grateful.

**Comment by modmuss50** [comment]
Me and @gigabit101 and @covers1624 would be willing to take over all of your mods.

**Comment by Chicken-Bones** [comment]
Fortunately, I just recently came back from the dead, and can confirm that there will be 1.8.8 ports of NEI, EnderStorage and ChickenChunks in the next week. I'm leaving FMP in the hands of McMultipart, and if you'd be willing to take over WR-CBE for the time being that'd be great.

**Comment by chloeprince** [comment]
Hey CB,

I know you contribute to Project Red, so why not just combine WR-CBE into Project Red? 

It seems like it really belongs with it.
Maybe Mr TJP could do it if you asked :)

**Comment by covers1624** [comment]
I like the idea of moving it to Project Red, but some people will want basic wireless redstone without the overhead of the rest of Project Red. if Mr TJP wants to add it to Project red then sweet it will be cool, but i will be making my own port of it for 1.8+

**Comment by chloeprince** [comment]
If you do update it, I hope that you will keep the aspect they work with Forge Multipart. For me, that is one of the beauties I love this mod so much, is that the transmitter and Receiver can be tucked into tight spots and covered over with Forge Microblocks.

**Comment by covers1624** [comment]
Definitely, except FMP will not be moving to 1.8 so i will be supporting McMultipart, Just waiting on CCL to be on the Forge Maven to start work on it.

**Comment by chloeprince** [comment]
That's great news. Thank you for taking up the cause. 

BTW,
What's the difference between McMultipart and FMP?

**Comment by covers1624** [comment]
McMultipart is for 1.8
Thats all i really know at this point, haven't looked in to it that much.

**Comment by covers1624** [comment]
Leaving this here in the hopes that CB catches this in his email.. https://github.com/TheCBProject
This is a thing... I am requesting permission from ChickenBones to develop this further and push releases of all of these. I am willing to do what ever i can to keep these mods alive. I really don't want to change the name of them so i can release them but i may have to, i feel that a name change will make them not the same.

**Comment by Chicken-Bones** [comment]
Permission granted

**Comment by covers1624** [comment]
Thank you, You are able to at anytime request that i pull the mods and i shall.

**Comment by MrTJP** [comment]
Ok, this is the first I'm hearing about this. I'm seriously considering taking over, I will never be comfortable using anything other than FMP/CCL.

@Chicken-Bones, open license aside, I'm still going to ask you directly. I am requesting permission to keep FMP alive.

Also, which one of your mods is still being worked on? CCL is a dependency to FMP, can't maintain one without the other...

**Comment by gigabit101** [comment]
MrTjp mcmultipart is going to be merged into forge soon I see know point in
having forge multipart and a multipart Api in forge also ccl and ccc are
already released for 1.9

**Comment by covers1624** [comment]
@MrTJP CCL for 1.9 can be found over on the project i linked and i am updating all his mods except FMP, if you want access to the CCL repo over there for updating things just ask and ill be happy to add you.

**Comment by Chicken-Bones** [comment]
@MrTJP Permission granted to do whatever you'd like. You've done a good job with Project Red and I'd expect no less from anything else.

**Comment by MrTJP** [comment]
Until I see a clear advantage in using MCMultipart over FMP (besides the fact that it is included in Forge), Project Red will remain using it as is. I'm going to try and not make too many changes to it in case CB decides to come back and wishes to pick it up.

Not really sure why the Forge team decided to go with that over FMP, especially since its scala codebase allows for so many things not even remotely possible with a Java implementation.. (again, no offense to amadornes)

**Comment by amadornes** [comment]
I'm not offended, don't worry :P
I agree that scala is way more powerful in some cases, yes, but the Forge team isn't (and has never been) willing to use anything different from what MC uses. That is, Java 6 (possibly Java 8 in 1.10?).
Since everything in MCMultiPart is built with modularity in mind and everything works with interfaces, soyou should be able to natively support it in your fork of FMP and have it be a scala implementation of the default API instead of its own fully independant thing. If I haven't derped anywhere, nothing in the API should force a dependency on a non-API class. That way you can continue using it and still have full interopability with what most other people use (I'm willing to help out with this, if you want).

**Comment by MrTJP** [comment]
If this is possible I will definitely look into it, so people wouldn't have to choose between the two APIs. 

Also, @Chicken-Bones, your runtime ASM mixin compiler is quite complex. If you could contact me and give me a brief overview on how it works, I would really appreciate it.

**Comment by whitestarx0312** [comment]
Just a random insignificant mod player over here like:
oh shit

**Comment by darthvader45** [comment]
@covers1624 Just wondering why the Minecraft updater keeps saying that NEI has an update available on the chickenbones.net site, yet I'm using the latest version available on that site for 1.11.2.  Any idea why that is?

**Comment by czd1999** [comment]
> Permission granted

Hello, I would like to use covers1624's version of chickenbonescore in a modpack. I don't know which of you to ask for permission so I'd like to ask both of you.  Is this okay?

**Comment by DemonReaper2000** [comment]
hi i have a question why is there now 1.7..10 update for not enough items i mean there was before from what i remember but now its only 1.8 and above so can you help me i really need a recipes mod for 1.7.10 ;-; TvT

**Comment by darthvader45** [comment]
Honestly, 1.7.10 is ancient now.  http://www.howoldisminecraft1710.today  I would get 1.12.2 and the version of NEI for that.  So much more now, more than 1.7.10 for sure.

**Comment by tomodachi94** [comment]
Hi @cdz1999 ,
Short answer: yes, you can use it in your modpack without permission, though I know modders like having their work attributed.
Long answer:

> Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions: The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

Translation from Legalese:

The MIT License allows you to distribute the "software" (aka the mod) for free, no explicit permissions necessary. (Though it is always nice to attribute the mod authors.)

EDIT: Kind of late, but better now then never.
ANOTHER EDIT: Sorry for accidentially tagging you, DemonReaper2000.
